### PR TITLE
Update go-txfile and dependencies

### LINKED ISSUE #7849: [libbeat] - Spool to disk permissions issues on Windows
When starting disk spooling it checks the permissions of the existing file and if they are more permissive that the configured `queue.spool.file.permissions` it results in an error.

https://github.com/elastic/beats/blob/36411a3616882547eb961cea13b7c2199bb31b3b/libbeat/publisher/queue/spool/spool.go#L82-L86

But Windows does not have standard POSIX file permissions since it uses a DACL. So we should skip this check for Windows. Take a look at how the mode bits get set on Windows in: https://github.com/golang/go/blob/4c77cb3087363f2ebab565bd2b46935749ca7dd2/src/os/types_windows.go#L36-L40

`
2018-08-02T12:09:17.067+0800 ERROR instance/beat.go:691 Exiting: error initializing publisher: file permissions must be more strict (required permissions: -rw-------, actual permissions: -rw-rw-rw-)
`

For confirmed bugs, please report:
- Version: 6.3.1
- Operating System: Windows 10
- Discuss Forum URL: https://discuss.elastic.co/t/metricbeat-fail-safe-over-function-on-output/142466
- Steps to Reproduce: 
- Workaround:
  - Set `queue.spool.file.permissions: 0777` (probably 0666 would work too).

### PR DESCRIPTION
**Pull request by urso**
Resolves #7849, #7867, #7720 

This PR updates go-txfile to v0.0.3 and golang.org/x/sys packages (go-txfile requires more recent version).

The update fixes a number of Windows related bugs in spooling to disk and replaces a call to fsync with fcntl+F_FULLFSYNC on darwin.

The error messages generated by go-txfile will be more detailed. We will further improve error handling and error reporting in the spool in follow up PRs.

The changes to Changelog.asciidoc for all user visible fixes and improvements.

### CODE REVIEW

**Comment by andrewkroh** [comment]
Can `TestProduceConsumer` be un-skipped with this change? That would resolve https://github.com/elastic/beats/issues/7720.

**Comment by urso** [comment]
@andrewkroh yes, it should be unskipped. Thank you!

File: libbeat/publisher/queue/spool/inbroker.go [hunk 1]
```diff
@@ -214,30 +221,32 @@ func (b *inBroker) eventLoop() {
...
 	if b.bufferedEvents == 0 {
 		return
 	}
 
-	// try to append pending events
+	// Try to flush pending events.
+	w := b.writer
 	for len(b.pending) > 0 {
-		n, err := b.queue.Writer().Write(b.pending)
+		n, err := w.Write(b.pending)
```
Comment (ph) [inline libbeat/publisher/queue/spool/inbroker.go]: This error will get hidden, should we log it as debug?

File: libbeat/publisher/queue/spool/outbroker.go [hunk 2]
```diff
@@ -81,6 +82,12 @@ var ackChanPool = sync.Pool{
 var errRetry = errors.New("retry")
 
 func newOutBroker(ctx *spoolCtx, qu *pq.Queue, flushTimeout time.Duration) (*outBroker, error) {
+	reader := qu.Reader()
+	avail, err := reader.Available()
+	if err != nil {
+		return nil, err
+	}
```
Comment (ph) [inline libbeat/publisher/queue/spool/outbroker.go]: Should we log it?

File: libbeat/publisher/queue/spool/inbroker.go (hunk 1 above)
Comment (urso) [reply libbeat/publisher/queue/spool/inbroker.go]: It's not logged on purpose. There can be a many reasons why this fails. E.g. IO error vs. spool is full. The later one is no error, but expected behavior. In future PRs we will check for the causes + handle/log them appropriately.

File: libbeat/publisher/queue/spool/outbroker.go (hunk 2 above)
Comment (urso) [reply libbeat/publisher/queue/spool/outbroker.go]: This error will be logged by the beat. The spool constructor fails if this call fails.

**Pull request merged**